src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -0,0 +1,2045 @@
+
+	//pressing a pad
+	if (on) {
+		//no need to pad press action if you've already processed it previously and pad was held
+		if (FXPress[xDisplay].yDisplay != yDisplay) {
+			backupPerformanceLayout();
+			//check if there a previously held press for this parameter in another column and disable it
+			//also transfer the previous value for that held pad to this new pad column press
+			for (int32_t i = 0; i < kDisplayWidth; i++) {
+				if (i != xDisplay) {
+					if ((layoutForPerformance[i].paramKind == lastSelectedParamKind)
+					    && (layoutForPerformance[i].paramID == lastSelectedParamID)) {
+						FXPress[xDisplay].previousKnobPosition = FXPress[i].previousKnobPosition;
+						initFXPress(&FXPress[i]);
+					}
+				}
+			}
+			padPressAction(modelStack, lastSelectedParamKind, lastSelectedParamID, xDisplay, yDisplay,
+			               !defaultEditingMode);
+		}
+	}
+	//releasing a pad
+	else {
+		//if releasing a pad with "held" status shortly after being given that status
+		//or releasing a pad that was not in "held" status but was a longer press and release
+		if ((isParamStutter(lastSelectedParamKind, lastSelectedParamID) && lastPadPress.isActive)
+		    || (FXPress[xDisplay].padPressHeld
+		        && ((AudioEngine::audioSampleTimer - FXPress[xDisplay].timeLastPadPress) < kHoldTime))
+		    || ((FXPress[xDisplay].previousKnobPosition != kNoSelection) && (FXPress[xDisplay].yDisplay == yDisplay)
+		        && ((AudioEngine::audioSampleTimer - FXPress[xDisplay].timeLastPadPress) >= kHoldTime))) {
+
+			padReleaseAction(modelStack, lastSelectedParamKind, lastSelectedParamID, xDisplay, !defaultEditingMode);
+		}
+		//if releasing a pad that was quickly pressed, give it held status
+		else if ((FXPress[xDisplay].previousKnobPosition != kNoSelection) && (FXPress[xDisplay].yDisplay == yDisplay)
+		         && ((AudioEngine::audioSampleTimer - FXPress[xDisplay].timeLastPadPress) < kHoldTime)) {
+			FXPress[xDisplay].padPressHeld = true;
+			logPerformanceLayoutChange();
+		}
+		updateLayoutChangeStatus();
+	}
+
+	//if you're in editing mode and not editing a param, pressing an FX column will open soundEditor menu
+	//if a parameter has been assigned to that FX column
+	if (defaultEditingMode && on) {
+		int32_t lastSelectedParamShortcutX = layoutForPerformance[lastPadPress.xDisplay].xDisplay;
+		int32_t lastSelectedParamShortcutY = layoutForPerformance[lastPadPress.xDisplay].yDisplay;
+
+		//if you're not already in soundEditor, enter soundEditor
+		if (getCurrentUI() != &soundEditor) {
+			goto potentialShortcutPadAction;
+		}
+		//if you're already in soundEditor, check if you're in the right menu
+		else if (soundEditor.getCurrentMenuItem()
+		         != paramShortcutsForSongView[lastSelectedParamShortcutX][lastSelectedParamShortcutY]) {
+			goto potentialShortcutPadAction;
+		}
+		//otherwise no need to do anything as you're already displaying the menu for the parameter
+		else {
+			return;
+		}
+potentialShortcutPadAction:
```

Comment:
Should use `||` in the `if` statements above and remove the gotos.

Reply:
Fixed thanks!